torch/testing/_internal/common_methods_invocations.py

```diff
@@ -430,11 +430,9 @@ def sample_inputs_slogdet(op_info, device, dtype, requires_grad):
         torch.randn(2, 0, 0, dtype=dtype, device=device),  # 'batched_0x0'
         torch.randn(2, S, S, dtype=dtype, device=device),  # 'batched_SxS'
     )
-    out = []
     for a in test_inputs:
         a.requires_grad = requires_grad
-        out.append(SampleInput(a))
-    return out
+        yield SampleInput(a)
```

Comment:
I prefer this change but we'll need to update the documentation for how sample inputs are generated to better explain what to do, too. In particular, we should clarify that these functions should "yield" their samples or return a generator expression (whether for a list, set, or tuple). 

A little aside from the main thrust of this PR is that we should also update our guidance on sample inputs to clarify that tensors and Python datastructures that are put into SampleInputs should be created each time the function is called. That is, calling iter() twice on the sample input function should return two distinct sets of objects, and modifying one set should not impact another. The reason I want to make this clear in the documentation is to avoid a pattern where, for example, someone declares a tensor and then always yields the same tensor. Consumers of sample inputs should be OK to modify them.

Does that sound good?

Reply:
Makes sense, I totally agree.